docs/proposals/201812_thanos-remote-receive.md

```diff
@@ -0,0 +1,174 @@
+
+## Architecture
+
+The Thanos receiver component seamlessly integrates into the rest of the Thanos components. It acts similarly to what is referred to in Thanos as a "source", in the current set of components, this is typically represented by the Thanos sidecar that is put next to Prometheus to ship tsdb blocks into object storage and reply to store API requests, however in the case of the Thanos receiver, the Thanos sidecar is not necessary anymore, as the data is replicated from the original Prometheus server to the Thanos receiver, and the Thanos receiver participates in the Thanos gossip mesh. The Prometheus server on a tenant’s infrastructure can therefore be completely vanilla and is just configured to replicate its time-series to the Thanos receiver.
+
+Instead of directly scraping metrics, however, the Thanos receiver accepts Prometheus remote write requests, and writes these into a local instance of the Prometheus tsdb, to prevent data leaking at the database level, each tenant has an individual tsdb instance, meaning a single Thanos receiver may manage multiple tsdb instances. When the receiver is queried, the respective tsdb is used to retrieve data. Furthermore, like the Thanos sidecar, the receiver answers Thanos store API requests and uploads built blocks of the Prometheus tsdb. Implementation-wise, this just requires wiring up existing components.
+
+In a minimal setup the system would look like the following:
+
+
+```
+                 +
+Tenant's Premise | Provider Premise
+                 |
+                 |            +------------------------+
+                 |            |                        |
+                 |  +-------->+     Object Storage     |
+                 |  |         |                        |
+                 |  |         +-----------+------------+
+                 |  |                     ^
+                 |  | S3 API              | S3 API
+                 |  |                     |
+                 |  |         +-----------+------------+
+                 |  |         |                        |       Store API
+                 |  |         |  Thanos Store Gateway  +<-----------------------+
+                 |  |         |                        |                        |
+                 |  |         +------------------------+                        |
+                 |  |                                                           |
+                 |  +---------------------+                                     |
+                 |                        |                                     |
++--------------+ |            +-----------+------------+              +---------+--------+
+|              | | Remote     |                        |  Store API   |                  |
+|  Prometheus  +------------->+     Thanos Receiver    +<-------------+  Thanos Querier  |
```

Comment:
When Thanos Receiver returns OK to Prometheus?
Once the data hits some TSDB instance? Earlier? Can we document this detail?

Reply:
Once the samples are successfully committed to all tsdb's, the request returns ok.

Reply:
:+1: